src/core/base64.cpp

```diff
@@ -54,4 +54,136 @@ QVariant decode( const QString & _b64, QVariant::Type _force_type )
 }
 
 
+//TODO C++20: It *may* be possible to make the following functions constepxr given C++20's constexpr std::string.
+
+/**
+ * @brief Base64 encodes data.
+ * @param data
+ * @return std::string containing the Base64 encoded data.
+ */
+auto encode(std::string_view data) -> std::string
+{
+	if (data.empty()) { return ""; }
+
+	// base64 encoded string
+	std::string result;
+
+	// number of chunks to process + padding
+	auto div_result = std::div(data.length(), numBytesPerChunk);
+	auto numChunks = div_result.quot;
+	auto numTrailingBytes = div_result.rem;
+
+	// add 1 chunk to handle last padded chunk
+	if (numTrailingBytes) { ++numChunks; }
+
+	// allocate one time to prevent in-loop reallocations
+	result.reserve(numChunks * numBase64CharPerChunk);
+
+	/*
+		char1               char2               char3
+		7 6 5 4 3 2   1 0 | 7 6 5 4   3 2 1 0 | 7 6   5 4 3 2 1 0
+		5 4 3 2 1 0 | 5 4   3 2 1 0 | 5 4 3 2   1 0 | 5 4 3 2 1 0
+		b64c1         b64c2           b64c3           b64c4
+	*/
+	// Get the first base64 character offset from the char chunk
+	auto b64char1 = [](std::string_view chunk) {
+		return static_cast<std::string::value_type>(chunk[0] >> 2);
```

Comment:
- Any reason for passing `std::string_view` by value?
- Given that `std::string` is an alias of `std::basic_string<char>`, `std::string::value_type` is always `char`.

Reply:
* My understanding of `string_view` was that this is the way to do it. If not then I can change it. 👍
* My thought was in the instance for some reason `std::string` isn't based on `char`.

Reply:
> My understanding of `string_view` was that this is the way to do it.

It turns out you're correct, indeed.